packages/nodes-from-anchor/src/v01/InstructionAccountNode.ts

```diff
@@ -17,26 +18,58 @@ import {
 import { IdlV01InstructionAccount, IdlV01InstructionAccountItem, IdlV01Seed } from './idl';
 import { pdaSeedNodeFromAnchorV01 } from './PdaSeedNode';
 
+function hasDuplicateAccountNames(idl: IdlV01InstructionAccountItem[]): boolean {
+    const seenNames = new Set<string>();
+
+    function checkDuplicates(items: IdlV01InstructionAccountItem[]): boolean {
+        for (const item of items) {
+            if ('accounts' in item) {
+                if (checkDuplicates(item.accounts)) {
+                    return true;
+                }
+            } else {
+                const name = camelCase(item.name ?? '');
+                if (seenNames.has(name)) {
+                    return true;
+                }
+                seenNames.add(name);
+            }
+        }
+        return false;
+    }
+
+    return checkDuplicates(idl);
+}
+
 export function instructionAccountNodesFromAnchorV01(
     idl: IdlV01InstructionAccountItem[],
     instructionArguments: InstructionArgumentNode[],
+    prefix?: string,
 ): InstructionAccountNode[] {
+    const hasDuplicates = hasDuplicateAccountNames(idl);
+    const shouldPrefix = prefix !== undefined || hasDuplicates;
```

Comment:
I think this should be like the v00 implementation to avoid checking for duplicates if we already have a prefix.


```suggestion
    const shouldPrefix = prefix !== undefined || hasDuplicateAccountNames(idl);
```